rest-server/engines/src/main/scala/com/actionml/engines/navhinting/NavHintingAlgorithm.scala

```diff
@@ -203,11 +214,19 @@ class NavHintingAlgorithm(dataset: NavHintingDataset)
 
   override def predict(query: NHQuery): NHQueryResult = {
     // find model elements that match eligible and sort by weight, sort before taking the top k
-    val results = query.eligibleNavIds.map((_,0d)).map { case (eligibleNavId, w) =>
-      dataset.navHintsDAO.findOneById(eligibleNavId).getOrElse(NavHint(eligibleNavId, 0))
-    }.filter(_.weight > 0).map { navHint => navHint._id -> navHint.weight } // swap key and value
-      .sortBy(-_._2) // sort by value, which is the score here, minus for
+    // adds the weights of hints with the same id from multiple conversion-id models
+    val results = query.eligibleNavIds.map((_,0d)).flatMap { case (eligibleNavId, w) =>
+      navHintsModels.map(_._2.findOneById(eligibleNavId))
+      //dataset.navHintsDAO.findOneById(eligibleNavId).getOrElse(NavHint(eligibleNavId, 0))
+    }.flatten // remove undefined Options
+      .groupBy(_._id).map { case (id, navHints) => // get nav-hints with matching ids
+        var sum = 0d
+        navHints.foreach(sum += _.weight)
+        NavHint(id, sum) // sum weights for all of same id
```

Comment:
sum here can also be `navHints.map(_.weight).sum` or `navHints.foldLeft(0)(_ + _.weight)`